Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/PermissionsHandler.java

```diff
@@ -0,0 +1,71 @@
+package org.dolphinemu.dolphinemu.utils;
+
+import android.annotation.TargetApi;
+import android.app.Activity;
+import android.app.AlertDialog;
+import android.content.DialogInterface;
+import android.content.pm.PackageManager;
+import android.os.Build;
+import android.support.v4.content.ContextCompat;
+import android.widget.Toast;
+
+import org.dolphinemu.dolphinemu.R;
+
+import static android.Manifest.permission.WRITE_EXTERNAL_STORAGE;
+
+public class PermissionsHandler {
+	public static final int REQUEST_CODE_WRITE_PERMISSION = 500;
+
+	@TargetApi(Build.VERSION_CODES.M)
+	public static boolean checkWritePermission(final Activity activity) {
+		if (android.os.Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+			return true;
+		}
+
+		int hasWriteContactsPermission = ContextCompat.checkSelfPermission(activity, WRITE_EXTERNAL_STORAGE);
```

Comment:
Yes, your are right. Because changing the location will also need the permission for the first stage since I plan to copy the files for those who had dolphin installed from the old location to the new location in the sdcard and not from the assets.

For new users we will copy from the assets, after a while we can stop that code and remove the write permission and all related code.